src/GovUk.Education.ExploreEducationStatistics.Public.Data.Model/Migrations/20240501113006_InitialMigration.cs

```diff
@@ -579,40 +580,67 @@ protected override void Up(MigrationBuilder migrationBuilder)
                 principalTable: "DataSetVersions",
                 principalColumn: "Id");
 
+            var isLocalEnvironment = 
+                Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") == Environments.Development;
+
             // Grant privileges on objects created by this resource's database user to the Admin App Service user.
-            var adminAppServiceIdentityName = Environment.GetEnvironmentVariable("AdminAppServiceIdentityName");
-            if (adminAppServiceIdentityName != null)
+            var adminAppServiceRoleName = isLocalEnvironment
```

Comment:
I know it's just grown this way, but this pattern repeats a few times in this file and is quite verbose:

```
            var adminAppServiceRoleName = isLocalEnvironment
                ? "app_admin" 
                : Environment.GetEnvironmentVariable("AdminAppServiceIdentityName");
            if (adminAppServiceRoleName != null)
            {
                migrationBuilder.Sql(
                    $"""GRANT SELECT, INSERT, UPDATE, DELETE ON ALL TABLES IN SCHEMA public TO {adminAppServiceRoleName}""");
                migrationBuilder.Sql(
                    $"""ALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT SELECT, INSERT, UPDATE, DELETE ON TABLES TO {adminAppServiceRoleName}""");
            }
```

I wonder if we can extract into a method that just takes the environment variable name and the PSQL role name?  Then we can very quickly reuse if and when we need more roles adding?

Reply:
I did do this and put these methods in `MigrationBuilderExtensions` but then with my most recent changes I don't think it's warranted anymore as there's now no need for so many occurrences of this, and there will be even less when we next squash all the migrations.

Just in case we need it again, this is what I was working on for the role names:

```csharp
private record AppService(string LocalPsqlRoleName, string AppIdentityEnvironmentVariableName)
{
    public static AppService AdminAppService => new("app_admin", "AdminAppServiceIdentityName");
    public static AppService DataProcessorFunctionApp => new("app_public_data_processor", "DataProcessorFunctionAppIdentityName");
    public static AppService PublisherFunctionApp => new("app_publisher", "PublisherFunctionAppIdentityName");

    private static bool IsLocalEnvironment => Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") == Environments.Development;

    public string PsqlRoleName => IsLocalEnvironment
        ? LocalPsqlRoleName
        : Environment.GetEnvironmentVariable(AppIdentityEnvironmentVariableName) ?? throw new Exception($"Environment variable {AppIdentityEnvironmentVariableName} is not set");
}
```

A role name could then be accessed like

```csharp
AppService.AdminAppService.PsqlRoleName
```

Reply:
Nice one!